controllers/ccruntime_controller.go

```diff
@@ -207,90 +206,107 @@ func (r *CcRuntimeReconciler) processCcRuntimeDeleteRequest() (ctrl.Result, erro
-				//
-				//        When looking at the kata-containers payload code, we see that the
-				//        the label is only set after containerd is successfully reconfigured,
-				//        and looking at this function we see we shouldn't reach this part
-				//        before the label is set.  However, that's not what we're facing ...
-				time.Sleep(time.Second * 60)
-
-				result, err = handlePostUninstall(r)
-				if !result.Requeue {
-					controllerutil.RemoveFinalizer(r.ccRuntime, RuntimeConfigFinalizer)
-					result, err = r.updateCcRuntime()
-					if err != nil {
-						return result, err
-					}
-					result, err = r.deleteUninstallDaemonsets()
-					prepostLabels := map[string]string{}
-					if r.ccRuntime.Spec.Config.PreInstall.Image != "" {
-						prepostLabels[PreInstallDoneLabel[0]] = PreInstallDoneLabel[1]
-					}
-					if r.ccRuntime.Spec.Config.PostUninstall.Image != "" {
-						prepostLabels[PostUninstallDoneLabel[0]] = PostUninstallDoneLabel[1]
+				result, err = r.updateCcRuntime()
+				if err != nil {
+					return result, err
+				}
 
-					}
-					err, nodes := r.getNodesWithLabels(prepostLabels)
-					if err != nil {
-						r.Log.Error(err, "an error occured when getting the list of nodes from which we want to"+
-							"remove preinstall/postuninstall labels")
-						return ctrl.Result{}, err
-					}
-					postuninstalledNodes := len(nodes.Items)
-					if !allNodesDone(postuninstalledNodes, r) {
-						return ctrl.Result{Requeue: true}, nil
-					} else {
-						if postuninstalledNodes > 0 {
-							result, err = r.removeNodeLabels(nodes)
-							if err != nil {
-								r.Log.Error(err, "removing the labels from nodes failed")
-								return ctrl.Result{}, err
-							}
+				result, err = r.deleteUninstallDaemonsets()
+				prepostLabels := map[string]string{}
+
+				if r.ccRuntime.Spec.Config.PreInstall.Image != "" {
+					prepostLabels[PreInstallDoneLabel[0]] = PreInstallDoneLabel[1]
+				}
+
+				if r.ccRuntime.Spec.Config.PostUninstall.Image != "" {
+					prepostLabels[PostUninstallDoneLabel[0]] = PostUninstallDoneLabel[1]
+
+				}
+
+				err, nodes := r.getNodesWithLabels(prepostLabels)
+				if err != nil {
+					r.Log.Error(err, "an error occured when getting the list of nodes from which we want to"+
+						"remove preinstall/postuninstall labels")
+					return ctrl.Result{}, err
+				}
+
+				postuninstalledNodes := len(nodes.Items)
+				if !allNodesDone(postuninstalledNodes, r) {
+					return ctrl.Result{Requeue: true}, nil
+				} else {
+					if postuninstalledNodes > 0 {
+						result, err = r.removeNodeLabels(nodes)
+						if err != nil {
+							r.Log.Error(err, "removing the labels from nodes failed")
+							return ctrl.Result{}, err
 						}
 					}
 				}
 			}
-			if err != nil {
-				return result, err
-			}
-			result, err = r.updateCcRuntime()
-			return result, err
 		}
 
-		result, err = r.updateUninstallationStatus(finishedNodes)
+		controllerutil.RemoveFinalizer(r.ccRuntime, RuntimeConfigFinalizer)
```

Comment:
Well moving the `RemoveFinalizer` to this common branch results in it being called twice when `r.ccRuntime.Spec.Config.PostUninstall.Image != ""` while previously it was only called once. Is repeating the call an issue or is it safe?

Reply:
I looked at the RemoveFinalizer code and it would not be a problem, but this was a nice catch. My goals was to this change as less disruptive as possible, so fixed with a force push. Thanks.